### DISCUSSION TITLE
J 4.x Adding Tags on the Fly with Parents

### DISCUSSION THREAD

**Discussion by coolcat-creations** [joomla/joomla-cms#47094] [2. Cold Features]
### Is your feature request related to a problem? Please describe.
When you have Tags with a Parent, they appear in the taglist like that:
Parent/Child 

but if you enter a Tag on the Fly like that it's not handled this way but is created like entered.

### Describe the solution you'd like
I would expect Joomla to create a Parent and a child

**Comment by brianteeman** [comment]
Not what I would expect. Maybe I am wrong but I dont recall ever seeing a UI where you can do this sort of thing

**Comment by coolcat-creations** [comment]
When I am in PHP Storm and create a new directory I can just type parent/child and it creates the parent folder and the child at once.

**Comment by crystalenka** [comment]
I think that this is common behavior when there is on-the-fly creation of items which can be nested. I've seen it in quite a few places, mostly filesystem related.

However, this could have unexpected consequences for people who are using tags that have the / as a part of the text, not an indication of nesting. Like "Bikes/bicycles". 

If we were to add it, auto-nested tags would have to be an opt-in behavior in my opinion, because it's specific to the mental model of developers or more systems-minded users. :) Would you agree?

**Comment by brianteeman** [comment]
Should also note that nested * is not always displayed the same way

![image](https://user-images.githubusercontent.com/1296369/208264577-8f19d410-e69b-44e8-983f-97ea9ddf1e86.png)

![image](https://user-images.githubusercontent.com/1296369/208264607-86b702ff-0452-437e-91f3-651e14c30440.png)

![image](https://user-images.githubusercontent.com/1296369/208264661-7a72103c-3c73-46ba-853b-b51d872fad45.png)

![image](https://user-images.githubusercontent.com/1296369/208264676-f6d1588a-a8b6-488a-a3e9-faf15da54d04.png)

**Comment by crystalenka** [comment]
oh, boy. Good point Brian, that's an issue in and of itself. 🙈

**Comment by brianteeman** [comment]
I only posted a few examples

**Comment by crystalenka** [comment]
Feel free to post a whole new issue about it and tag me.

**Comment by brianteeman** [comment]
If this is to be implemented as a new feature then it will need to be consistently applied throughout joomla

**Comment by coolcat-creations** [comment]
I would love to see here rather more consistency and fix the appearance of the nesting. That would remove the confusion that I could do something like parent/child as well. Could someone remove the new feature tag and add a UX tag instead?

**Comment by brianteeman** [comment]
You need to be very careful here. Whatever you display has to make sense when you are filtering the results

**Comment by coolcat-creations** [comment]
But that's the same issue for categories or not?

**Comment by Hackwar** [comment]
You are actually supposed to be able to switch between "ajax" and "nested" mode. This is indeed another inconsistency in the tags component. I would like to introduce an option to limit tags only to one level, which would mean URLs like /tags/[tag-1]/[tag-2]/..., or nested tags, which would mean /tags/[parent-tag]/[child-level1]/[child-level2]/...

**Comment by coolcat-creations** [comment]
Actually, I use nested tags a lot, and it's not good to limit them. Or did I misunderstand you?

**Comment by Hackwar** [comment]
Yes, you misunderstood me.

**Comment by bascherz** [comment]
Sure hope this ticket is still open and of relatively high priority.

What I am seeing with Parent/Child tags (v5.4.2) is that in the Publishing tab of an article, not all child tags are listed. For instance, I have around 20 tags in a group named "Methods" and only a random 4 of them show up in the dropdown list. Aren't all existing tags supposed to be shown when you click in that field? If I type "Methods", all child tags then appear. But the problem with this is when you have a lot of tags (which this site does!) you have to remember the parent or child tag name. And if the tag is a child, you type it but you don't notice it is shown as parent/child and you press Enter, you create a new top-level tag with that name that is already a child. The list should be complete upon clicking!

Correction to the above: All of them show up, just not together as a "group" having the same parent. That's very non-intuitive. Oh, and that is in AJAX tag entry mode. In Nested mode, yes they are nested and in alphabetical order, but if you type a parent tag name in that mode, you don't see any of its child tags.
